### PR TITLE
Refactor/remove rhytmic value

### PR DESCRIPTION
**Pull request by JakubBilski**
I removed `sound.rhythmic_value` from everywhere. Backend no longer operates on "2." rhythmic value notation. Now, every baseSound has `duration` property, which means "how many 32s are equivalent to the rhythmic value of this baseSound". In my opinion, after these changes quality of code increased dramatically.

I did not change `update_sounds_with_rhythmic_values_compare_absolute` method names, because I think that `update_sounds_with_durations_compare_absolute` is a bad name. Maybe some third option?

I checked chords generation of all the three methods using "ach_spij_C.wav", and it seems everything works just as before refactor.

I also checked sound generation, and, unfortunately, scores dropped about 4 points each, and I don't know why.

### CODE REVIEW

File: src/chords_generation/daria.py [hunk 1]
```diff
@@ -105,14 +98,14 @@ def get_chords_daria(sounds: List[music.Sound],
                 max_chord = c
 
         t_c.append(music.Chord(max_chord.note,
-                               duration=half_bar_len/4,
+                               duration=constants.MIN_CHORD_DURATION,
```
Comment (JakubBrojacz) [inline src/chords_generation/daria.py:101]: what is MIN_CHORD_DURATION?
This should be equal to width of window used for scoring.
This changes output of algorithm in, for example, lato_lato_C
left - before changes
right - after changes
obraz

File: src/chords_generation/daria.py [hunk 2]
```diff
@@ -105,14 +98,14 @@ def get_chords_daria(sounds: List[music.Sound],
...
                                kind=max_chord.kind))
         last_chord_level = max_chord.level
-        i += half_bar_len
+        i += constants.MIN_CHORD_DURATION
```
Comment (JakubBrojacz) [inline src/chords_generation/daria.py:104]: `i` should also be changed by width of the window

File: src/chords_generation/hausner.py [hunk 3]
```diff
@@ -91,12 +85,12 @@ def get_chords_hausner(sounds: List[music.Sound],
                 max_score = score
                 max_chord = c
         t_c.append(music.Chord(max_chord.note,
-                               duration=half_meter_len/4,
+                               duration=constants.MIN_CHORD_DURATION,
                                kind=max_chord.kind))
-        i += half_meter_len
+        i += constants.MIN_CHORD_DURATION
```
Comment (JakubBrojacz) [inline src/chords_generation/hausner.py:90]: same here

File: src/chords_generation/hausner.py [hunk 4]
```diff
@@ -91,12 +85,12 @@ def get_chords_hausner(sounds: List[music.Sound],
                 max_score = score
                 max_chord = c
         t_c.append(music.Chord(max_chord.note,
-                               duration=half_meter_len/4,
+                               duration=constants.MIN_CHORD_DURATION,
```
Comment (JakubBrojacz) [inline src/chords_generation/hausner.py:88]: same here

File: src/perfect_sounds_creation.py [hunk 5]
```diff
@@ -1,32 +1,10 @@
 from music import Sound
 
-def get_perfect_sounds(sounds, beat):
-    update_beat_fractions_from_rhythmic_values(sounds)
+def get_perfect_sounds(sounds, beat_duration, beat_duration_ms):
```
Comment (JakubBilski) [inline src/perfect_sounds_creation.py:3]: This is a little inconsistent - here beat_duration, beat_duration_ms is used to determine duration_ms of sounds, while in create_midi it is duration_ms_of_32. I think the latter is better. What do you think?

File: src/main.py [hunk 6]
```diff
@@ -61,7 +61,7 @@ def frontend_communication():
         "notes": [
             {
                 "symbol": note.symbol,
-                "rhythmic_value": note.rhythmic_value
+                "duration": note.duration
```
Comment (JakubBrojacz) [inline src/main.py:64]: we need to return rhythmic value (convert from duration to rhythmic value here)

File: src/main.py [hunk 7]
```diff
@@ -129,9 +129,10 @@ def process_file(filename):
     else:
         tonation = tonation_recognition.get_tonation(sounds)
 
-    chords = chords_generation.get_chords_daria(sounds, tonation, (4, 4))
+    chords = chords_generation.get_chords_hausner(sounds, tonation, (4, 8))
```
Comment (JakubBrojacz) [inline src/main.py:132]: change to daria algorithm before merging

File: src/music/chord.py [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
+from utils import constants
 
 class Chord(base_sound.BaseSound):
     def __init__(self, note=None, symbol=None, duration=None,
                  kind=None):
         super().__init__(note, symbol)
-        self.kind = kind
-        self.duration = duration
+        if kind is not None:
+            self.kind = kind
```
Comment (JakubBrojacz) [inline src/music/chord.py:9]: Is it better that chord doesn't have property `kind` than it has property `kind` equal `None`?

File: src/music/chord.py [hunk 9]
```diff
@@ -45,3 +47,9 @@ def __str__(self):
 
     def __eq__(self, other):
         return (self.note == other.note) and (self.kind == other.kind)
+
+    @base_sound.BaseSound.duration.setter
+    def duration(self, duration):
+        if isinstance(duration, float):
+            raise Exception(f"Tried to set duration to float")
+        self._duration = duration
```
Comment (JakubBrojacz) [inline src/music/chord.py:55]: No endline

File: src/perfect_sounds_creation.py [hunk 10]
```diff
@@ -1,32 +1,10 @@
 from music import Sound
```
Comment (JakubBrojacz) [inline src/perfect_sounds_creation.py:1]: We can remove this file

File: src/tests/sound_from_file_test.py [hunk 11]
```diff
@@ -90,22 +91,18 @@ def visualize_d(d, sounds, test_sounds):
...
-            ds = str(int(int(ds)*coef))
-        except Exception:
-            ds = ""
-        if d:
-            ds = f"{ds}."
+        if ds not in constants.LEGAL_NOT_DOTTED_DURATION_VALUES:
+            ds = (ds//3)*2
+        ds = int(ds*coef)
+        if ds not in constants.LEGAL_DURATION_VALUES:
+            return None
```
Comment (JakubBrojacz) [inline src/tests/sound_from_file_test.py:103]: You missed the part with remembering the dot in `d=True` line. This few lines just multiplies note by coef (with respect to dotted notes).

File: src/tests/test_data.py [hunk 12]
```diff
@@ -17,6 +17,20 @@ def get_all_test_models():
     models = get_other_rec_test_models()
     return models
 
+def rhythmic_value_to_duration(rhythmic_value):
```
Comment (JakubBrojacz) [inline src/tests/test_data.py:20]: Is can be in utils.constants

**Review by JakubBrojacz — CHANGES_REQUESTED** [review]

File: src/chords_generation/daria.py (hunk 1 above)
Comment (JakubBilski) [reply src/chords_generation/daria.py:101]: Yes, this is a parameter that changes the behaviour of the algorithm. It should be in the daria.py file indeed

File: src/music/chord.py (hunk 8 above)
Comment (JakubBilski) [reply src/music/chord.py:9]: It's better when you don't know if there is any setter in the base class that will throw an exception when trying to set None

File: src/tests/test_data.py (hunk 12 above)
Comment (JakubBilski) [reply src/tests/test_data.py:20]: I disagree, that would mean that we connect output format for frontend with input format for tests

Comment (JakubBrojacz) [reply src/tests/test_data.py:20]: Yeah, those are the same formats

File: src/music/chord.py (hunk 8 above)
Comment (JakubBrojacz) [reply src/music/chord.py:9]: ok

File: src/tests/sound_from_file_test.py (hunk 11 above)
Comment (JakubBilski) [reply src/tests/sound_from_file_test.py]: I see you fixed this

Comment (JakubBrojacz) [reply src/tests/sound_from_file_test.py]: I fixed

File: src/chords_generation/daria.py (hunk 2 above)
Comment (JakubBilski) [reply src/chords_generation/daria.py]: Done

File: src/chords_generation/daria.py (hunk 1 above)
Comment (JakubBilski) [reply src/chords_generation/daria.py]: Done

File: src/chords_generation/hausner.py (hunk 3 above)
Comment (JakubBilski) [reply src/chords_generation/hausner.py]: Done

File: src/chords_generation/hausner.py (hunk 4 above)
Comment (JakubBilski) [reply src/chords_generation/hausner.py]: Done

File: src/music/chord.py (hunk 9 above)
Comment (JakubBilski) [reply src/music/chord.py]: Fixed

File: src/perfect_sounds_creation.py (hunk 10 above)
Comment (JakubBilski) [reply src/perfect_sounds_creation.py]: Removed

File: src/tests/test_data.py (hunk 12 above)
Comment (JakubBilski) [reply src/tests/test_data.py]: Done

File: src/main.py (hunk 6 above)
Comment (JakubBilski) [reply src/main.py]: Done

**Review by JakubBrojacz — APPROVED** [review]

**Pull request merged**